src/pyfmi/fmi3.pyx

```diff
@@ -886,6 +1111,174 @@ cdef class FMUModelME3(FMUModelBase3):
+        status = FMIL3.fmi3_import_get_nominals_of_continuous_states(self._fmu, xnominal, nx)
+        self._log_handler.capi_end_callback(self._max_log_size_msg_sent, self._current_log_size)
+        return status
+
+    def _get_nominal_continuous_states(self):
+        """
+        Returns the nominal values of the continuous states.
+
+        Returns::
+            The nominal values.
+        """
+        cdef FMIL3.fmi3_status_t status
+        cdef np.ndarray[FMIL3.fmi3_float64_t, ndim=1, mode='c'] xn = np.zeros(self._nContinuousStates, dtype=np.double)
+
+        if self._initialized_fmu == 0:
+            raise FMUException("Unable to retrieve nominals of continuous states, FMU must first be initialized.")
+
+        status = self._get_nominal_continuous_states_fmil(<FMIL3.fmi3_float64_t*> xn.data, self._nContinuousStates)
+        if status != 0:
+            raise FMUException('Failed to get the nominal values.')
+
+        # Fallback - auto-correct the illegal nominal values:
+        xnames = list(self.get_states_list().keys())
+        for i in range(self._nContinuousStates):
+            if xn[i] == 0.0:
+                if self.callbacks.log_level >= FMIL.jm_log_level_warning:
+                    logging.warning(f"The nominal value for {xnames[i]} is 0.0 which is illegal according " + \
+                                     "to the FMI specification. Setting the nominal to 1.0.")
+                xn[i] = 1.0
+            elif xn[i] < 0.0:
+                if self.callbacks.log_level >= FMIL.jm_log_level_warning:
+                    logging.warning(f"The nominal value for {xnames[i]} is <0.0 which is illegal according " + \
+                                    f"to the FMI specification. Setting the nominal to abs({xn[i]}).")
+                xn[i] = abs(xn[i])
+
+        return xn
+
+    nominal_continuous_states = property(_get_nominal_continuous_states, doc =
+    """
+    Property for accessing the nominal values of the continuous states. Calls
+    the low-level FMI function: fmi3GetNominalContinuousStates.
+    """)
+
+    cdef FMIL3.fmi3_status_t _completed_integrator_step(self,
+            FMIL3.fmi3_boolean_t no_set_FMU_state_prior_to_current_point,
+            FMIL3.fmi3_boolean_t* enter_event_mode,
+            FMIL3.fmi3_boolean_t* terminate_simulation
+        ):
+        cdef FMIL3.fmi3_status_t status
+
+        self._log_handler.capi_start_callback(self._max_log_size_msg_sent, self._current_log_size)
+        status = FMIL3.fmi3_import_completed_integrator_step(
+            self._fmu,
+            no_set_FMU_state_prior_to_current_point,
+            enter_event_mode,
+            terminate_simulation
+        )
+        self._log_handler.capi_end_callback(self._max_log_size_msg_sent, self._current_log_size)
+
+        self._last_accepted_time = self._get_time()
+        return status
+
+    def completed_integrator_step(self, no_set_FMU_state_prior_to_current_point = True):
```

Comment:
Any particular reason for the default value here?

Plus should probably add a docstring for the new parameter

Reply:
I couldn't see why either, but it is the same in FMI2 and I didn't want to change the interface.